File: f2py_example.py
import fort_example as fe
import numpy as np

# run a simple test.
p = (4, 5, 1)
print('{:.1f} * {:.1f} + {:.1f} = {:.1f}'.format(*p, fe.test_module.multiply_add(*p)))

p = (
    np.asfortranarray([[1, 0], [0, 1]]),
    np.asfortranarray([[2, 0], [0, 2]]),
    np.asfortranarray([[0, 1], [1, 0]])
)
print(fe.test_module.multiply_add_mat(*p))

File: f2py_example.f90
module test_module
implicit none
contains

subroutine multiply_add(a, b, c, d)
    double precision, intent(in) :: a, b, c
    double precision, intent(out) :: d

    d = a*b + c

end subroutine multiply_add

subroutine multiply_add_mat(a, b, c, d)
    double precision, dimension(2,2), intent(in) :: a, b, c
    double precision, dimension(2,2), intent(out) :: d

    d = matmul(a, b) + c

end subroutine multiply_add_mat

end module test_module
